build-osx.sh

```diff
@@ -1,17 +1,208 @@
+}
+
+run_premake_if()
+{
+    if [[ premake5.lua -nt Surge.xcworkspace/premake-stamp ]]; then
+        run_premake
+    fi
+}
+
+run_clean()
+{
+    flavor=$1
+    xcodebuild clean -configuration Release -project surge-${flavor}.xcodeproj
+}
+
+run_build()
+{
+    flavor=$1
+    xcodebuild build -configuration Release -project surge-${flavor}.xcodeproj
+}
+
+default_action()
+{
+    run_premake
+    if [ -n "$VST2SDK_DIR" ]; then
+        run_clean "vst"
+        run_build "vst"
+    fi
+
+    run_clean "vst3"
+    run_build "vst3"
+
+    run_clean "au"
+    run_build "au"
+}
+
+run_all_builds()
+{
+    run_premake_if
+    if [ -n "$VST2SDK_DIR" ]; then
+        run_build "vst"
+    fi
+
+    run_build "vst3"
+    run_build "au"
+}
+
+run_install_local()
+{
+    rsync -r "resources/data/" "$HOME/Library/Application Support/Surge/"
+
+    if [ -n "$VST2SDK_DIR" ]; then
+        rsync -r --delete "products/Surge.vst/" ~/Library/Audio/Plug-Ins/VST/Surge.vst/
+    fi
+    
+    rsync -r --delete "products/Surge.component/" ~/Library/Audio/Plug-Ins/Components/Surge.component/
+    rsync -r --delete "products/Surge.vst3/" ~/Library/Audio/Plug-Ins/VST3/Surge.vst3/
+}
+
+run_build_validate_au()
+{
+    run_premake_if
+    run_build "au"
+
+    rsync -r "resources/data/" "$HOME/Library/Application Support/Surge/"
+    rsync -r --delete "products/Surge.component/" ~/Library/Audio/Plug-Ins/Components/Surge.component/
+
+    auval -vt aumu VmbA
+}
+
+run_clean_builds()
+{
+    run_premake_if
+    if [ -n "$VST2SDK_DIR" ]; then
+        run_clean "vst"
+    fi
+
+    run_clean "vst3"
+    run_clean "au"
+}
+
+run_clean_all()
+{
+    run_clean_builds
+    rm -rf Surge.xcworkspace *xcodeproj target product
+}
+
+run_uninstall_surge()
+{
+    sudo rm -rf /Library/Audio/Plug-Ins/Components/Surge.component
+    sudo rm -rf /Library/Audio/Plug-Ins/VST/Surge.vst
+    sudo rm -rf /Library/Audio/Plug-Ins/VST3/Surge.vst3
+    sudo rm -rf ~/Library/Audio/Plug-Ins/Components/Surge.component
+    sudo rm -rf ~/Library/Audio/Plug-Ins/VST/Surge.vst
+    sudo rm -rf ~/Library/Audio/Plug-Ins/VST3/Surge.vst3
+    sudo rm -rf ~/Library/Application\ Support/Surge
+}
+
+# This is the main section of the script
+command="$1"
+
+# put help up here so it runs even without the prerequisite check
+if [ "$command" = "--help" ]; then
+    help_message
+    exit 0
 fi
-xcodebuild clean -project surge-vst3.xcodeproj
-xcodebuild build -configuration Release -project surge-vst3.xcodeproj
-xcodebuild clean -project surge-au.xcodeproj
-xcodebuild build -configuration Release -project surge-au.xcodeproj
+
+prerequisite_check
+
+# if you add a Key here then you need to implement it as a function and add it to the help above
+# to get the PR swept. Thanks!
+case $command in
+    --premake)
+        run_premake
+        ;;
+    --build)
+        run_all_builds
+        ;;
+    --install-local)
+        run_install_local
+        ;;
+    --build-and-install)
```

Comment:
nice, this is what I was looking for; seems the most generally useful case for development